pages/views/elements.py

```diff
@@ -218,3 +226,44 @@ def get_context_data(self, **kwargs):
                         kwargs=url_kwargs),
                 })
         return context
+
+
+class CertificateDownloadView(LoginRequiredMixin, DetailView):
+    model = Sequence
+    slug_url_kwarg = 'sequence_slug'
+    template_name = 'pages/certificate.html'
+    response_class = PdfTemplateResponse
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        sequence = self.object
+        sequence_progress = get_object_or_404(
+            SequenceProgress,
+            sequence=sequence,
+            user=self.request.user)
+        enumerated_elements = EnumeratedElements.objects.filter(
+            sequence=sequence)
+        user_enumerated_progress = EnumeratedProgress.objects.filter(
+            progress=sequence_progress)
+
+        has_completed_sequence = True
+        for element in enumerated_elements:
+            user_element_progress = user_enumerated_progress.filter(rank=element.rank).first()
```

Comment:
Relies on N (maximum number of elements) queries to the database for comparing two sequences.